src/components/Login/LoginContainer.tsx

```diff
@@ -91,6 +99,59 @@ export const InitialisationContainer: FunctionComponent<NavigationProps> = ({
     setLoginStage("SCAN");
   };
 
+  const shouldResendAlert = async (): Promise<boolean> => {
+    return new Promise(resolve => {
+      Alert.alert(
+        "Resend OTP?",
+        lastResendWarningMessage,
+        [
+          {
+            text: "CANCEL",
+            onPress: () => resolve(false)
+          },
+          {
+            text: "RESEND",
+            onPress: () => resolve(true)
+          }
+        ],
+        { cancelable: false }
+      );
+    });
+  };
+
+  const handleRequestOTP = async (fullNumber?: string): Promise<boolean> => {
+    try {
+      fullNumber = fullNumber ?? mobileNumber;
+      if (lastResendWarningMessage) {
```

Comment:
Do we need this condition? Can we move this into the `shouldResendAlert`?

I'm thinking the concept of the `shouldResendAlert` is closer to `getResendConfirmationIfNeeded`, so the consumer just calls this and if it's true, proceed.

Reply:
i see. that sounds better, thanks!!